pkg/controllers/user/setting/register.go

```diff
@@ -0,0 +1,173 @@
+package setting
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/rancher/types/apis/core/v1"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// SettingProviderController listens for setting CUD in management API
+// and propagates the changes to the secret of cattle-system namespace in cluster API
+type Controller struct {
+	secrets                 v1.SecretInterface
+	secretsLister           v1.SecretLister
+	managementSettingLister v3.SettingLister
+	clusterName             string
+}
+
+const (
+	CattleNamespace          = "cattle-system"
+	DefaultSettingSecretName = "cattle-setting-provider"
+	settingSecretAnnotation  = "secret.setting.cattle.io/secret"
+	LocalClusterName         = "local"
+)
+
+var syncSettings = []string{"ingress-ip-domain", "rdns-base-url"}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	c := &Controller{
+		secrets:                 cluster.Core.Secrets(""),
+		secretsLister:           cluster.Core.Secrets("").Controller().Lister(),
+		managementSettingLister: cluster.Management.Management.Settings("").Controller().Lister(),
+		clusterName:             cluster.ClusterName,
+	}
+
+	if cluster.ClusterName == LocalClusterName {
+		return
+	}
+
+	sync := v3.NewSettingLifecycleAdapter(fmt.Sprintf("settingProviderController_%s", cluster.ClusterName), true,
+		cluster.Management.Management.Settings(""), c)
+
+	cluster.Management.Management.Settings("").AddHandler(ctx, "settingProviderController",
+		func(key string, obj *v3.Setting) (runtime.Object, error) {
+			if obj == nil {
+				return sync(key, nil)
+			}
+
+			// only propagate settings within the sync settings array
+			_, found := find(syncSettings, obj.Name)
+			if !found {
+				return nil, nil
+			}
+
+			if obj.Labels != nil {
+				if obj.Labels["cattle.io/creator"] == "norman" {
+					return sync(key, obj)
+				}
+			}
+
+			return nil, nil
+		})
+}
+
+func (c *Controller) Create(obj *v3.Setting) (runtime.Object, error) {
```

Comment:
I'd suggest we use sync instead of lifecycle. Lifecycles are heavy, and should be used with caution